app/lib/analyzer/task_sources.dart

```diff
@@ -4,24 +4,118 @@
 
 import 'dart:async';
 
+import 'package:async/async.dart';
+import 'package:gcloud/db.dart';
+
+import '../frontend/models.dart';
 import '../shared/task_scheduler.dart';
 
+import 'models.dart';
+
+/// Creates a task when a version uploaded in the past hour has no analysis yet.
 class DatastoreHeadTaskSource extends PollingTaskSource {
-  DatastoreHeadTaskSource() : super(new Duration(minutes: 10));
+  final DatastoreDB _db;
+  DateTime _lastTs;
+  DatastoreHeadTaskSource(this._db) : super(const Duration(minutes: 10));
 
   @override
   Stream<Task> pollTasks() {
-    // TODO: implement pollTasks
-    return new Stream.empty();
+    final DateTime now = new DateTime.now().toUtc();
+    final DateTime oneHourAgo = now.subtract(new Duration(hours: 1));
+    DateTime minCreated;
+    if (_lastTs == null) {
+      minCreated = oneHourAgo;
+    } else if (_lastTs.isBefore(oneHourAgo)) {
+      // more than an hour passed since the last poll
+      minCreated = _lastTs;
+    } else {
+      minCreated = oneHourAgo;
+    }
```

Comment:
Not quite sure I understand your description.

Every single newly uploaded `PackageVersion` will have a `created` timestamp. This gives us a total utc timestamp order. The probability of two newly-uploaded packages having the very same timestamp is very unlikely. So we just remember the very last `PackageVersion.created` we analyzed and continue the next query from there.

The only two possible issues are: two `PackageVersion`s have the exact same `created` timestamp or there is a clock difference where one VM's clock is ahead of another one and it basically creates a `PackageVersion` with an wrong/too-old `created` timestamp.

Anyways, I'm fine with the 1 hour approach, if you prefer it.